cmd/tkn/main.go

```diff
@@ -26,7 +29,25 @@ func main() {
 	tp := &cli.TektonParams{}
 	tkn := cmd.Root(tp)
 
-	if err := tkn.Execute(); err != nil {
+	args := os.Args[1:]
+	cmd, _, _ := tkn.Find(args)
+	if cmd != nil && cmd == tkn {
+		pluginCmd := "tkn-" + os.Args[1]
+		exCmd, err := exec.LookPath(pluginCmd)
+		if err != nil {
+			// Can't find the binary in PATH, bailing to usual execution
+			goto CoreTkn
+		}
+
+		if err := syscall.Exec(exCmd, append([]string{exCmd}, os.Args[2:]...), os.Environ()); err != nil {
+			fmt.Printf("Command finished with error: %v", err)
+			os.Exit(254)
+		}
+		return
+	}
+
+CoreTkn:
+	if err := cmd.Execute(); err != nil {
```

Comment:
won't this crash if `cmd == nil` ?

Reply:
@sthaha right, it will, but `cmd` is never `nil`. But this is an *artefact* of a refactoring I tried :sweat_smile: it should be `tkn.Execute()`, I'll update :sweat_smile: